DirectProgramming/C++SYCL_FPGA/Tutorials/DesignPatterns/memory_system/src/memory_system_kernels.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include "memory_system_defines.hpp"
+
+struct NaiveKernel {
+  void operator()() const {
+    SimpleInputT input;
+    SimpleOutputT output;
+
+    [[intel::fpga_memory("BLOCK_RAM")]]
+    int buffer[kNumRows][kNumCols];
```

Comment:
```suggestion
    int array2d[kNumRows][kNumCols];
```

I think the name `buffer` may be confused with SYCL buffers